File: backend/app/schemas/container.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID

class ContainerCreate(BaseModel):
    image_id: UUID
    name: Optional[str] = None

class ContainerResponse(BaseModel):
    id: str
    name: str
    image_name: str
    status: str
    vnc_url: Optional[str] = None
    ssh_command: Optional[str] = None
    created_at: datetime
    last_activity: datetime

    class Config:
        from_attributes = True

class ContainerStats(BaseModel):
    container_id: str
    cpu_usage: float
    memory_usage_mb: int
    storage_usage_gb: float
    timestamp: datetime

File: frontend/src/types/container.ts
export interface Container {
  id: string;
  name: string;
  image_name: string;
  status: string;
  vnc_url?: string;
  ssh_command?: string;
  created_at: string;
  last_activity: string;
  cpu_usage?: number;
  memory_usage_mb?: number;
}

export interface ContainerImage {
  id: string;
  name: string;
  description: string;
  category: string;
  cpu_limit: number;
  memory_limit_mb: number;
  storage_limit_gb: number;
}
